### PR TITLE
Resolve Display Issue of Install Add-On Buttons in Formidable Forms (Lite Version PR)

### PR DESCRIPTION
**Pull request by shervElmi**
### Main PR: https://github.com/Strategy11/formidable-pro/pull/4726 ###

### CODE REVIEW

**Review by garretlaxton — APPROVED** [review]
This looks great!

Something I do notice is that if the plugin is installed, but not activated it shows the same message that the plugin is not installed. Maybe we could improve that in the future?
image

**Comment by shervElmi** [comment]
Thanks Garret for your comment. @Crabcyborg, what are your thoughts on Garret's comment here? It's outside of this PR, but I would like to know if we need to create a new issue for that or not.

File: js/formidable_admin.js [hunk 1]
```diff
@@ -8596,11 +8596,12 @@ function frmAdminBuildJS() {
 			success: function( response ) {
 				let saveAndReload;
 
-				if ( 'string' !== typeof response && 'string' === typeof response.message ) {
-					if ( 'undefined' !== typeof response.saveAndReload ) {
-						saveAndReload = response.saveAndReload;
+				if ( 'string' !== typeof response && 'string' === typeof response?.data.message ) {
```
Comment (Crabcyborg) [inline js/formidable_admin.js:8599]: @shervElmi This is causing issues when I activate an add-on from the builder page.

My response is now in this new `response.data` format. It is at `response.message`.

Screen Shot 2024-01-23 at 9 27 34 AM

This results in this TypeError,
Screen Shot 2024-01-23 at 9 28 05 AM

**Comment by shervElmi** [comment]
@Crabcyborg, did you use the `fix-addon-buttons-display-issue` branch for the Formidable Pro plugin?

**Comment by Crabcyborg** [comment]
Thanks @garretlaxton and @shervElmi.

@stephywells Do you think it's worth trying to conditionally change the pop up title and message content based on state of the button? If it says "Activate", we could change it to say something like "This plugin is not activated. Would you like to activate it now?".

image

**Comment by Crabcyborg** [comment]
@shervElmi Yes I did. That update has no changes in it though that would change the response to include a `data` key though.

**Comment by Crabcyborg** [comment]
@shervElmi Can we make it work as well if Pro isn't up to date? It would be nice not to have these errors.

**Comment by shervElmi** [comment]
@Crabcyborg, should this PR be retested by Garret? Made some refactoring in my last commit.

**Comment by stephywells** [comment]
> Do you think it's worth trying to conditionally change the pop up title and message content based on state of the button? If it says "Activate", we could change it to say something like "This plugin is not activated. Would you like to activate it now?".

It would be much better if the text is all applicable for the situation (install vs activate). I know this has come up several times in the past, but I'm not sure if we created an issue for it.

**Comment by Crabcyborg** [comment]
Thanks Steph!

@shervElmi Can you make an issue for it?

**Comment by shervElmi** [comment]
@Crabcyborg, here is the issue URL: https://github.com/Strategy11/formidable-pro/issues/4745

**Comment by Crabcyborg** [comment]
Thank you Sherv!

**Review by Crabcyborg — APPROVED** [review]
Thanks @shervElmi! This looks good.

🚀

**Pull request merged**